### PR TITLE
[MDB Ignore]Adds helper for wallitems with dirs. A new Standard for mapping going forwards

### PR DESCRIPTION
**Pull request by Tsar-Salat**
## About The Pull Request

Adds dir mapping helpers from tg for air alarms, cameras, and other wallitems. Converts map dir varedits to actual pathing changes for consistency.

Adds emissives for air alarms

Adds autonaming to request consoles

TG coder had a spiel about `inverse` var shitcode. Thats not really why Im porting this, but its important so I will include it anyway.

> Wall items mostly use the direction from the floor to the wall in the named mapping helper. Wall items mostly use the direction from the wall to the floor for the internal dir variable.
> 
> This leads to a headache when it comes to working out what conflicts with what, and what needs placing where.
> 
> Wall frames provided a member, inverse, which specified whether or not to invert the direction of the item when looking for conflicts. It was also used to specify whether to look for conflicts outside of the wall (cameras and lights appear external to the wall) or inside the wall (most wall items). This flag was set for Intercoms, APCs, and Lights. Since APCs and Lights expect a floor-to-wall direction, and Intercoms expect a wall-to-floor direction, this means that APCs and Lights were getting the correct direction, and Intercoms were getting the wrong direction.
> 
> Some implications of this setup were:
> 
> You could build an APC on top of another wall item, provided there was nothing external attached to the wall and the area didn't have an APC.
> You could stack Intercoms indefinitely on top of the same wall, provided you weren't in a one-tile wide corridor with something on the opposite wall.

Ports:
- https://github.com/tgstation/tgstation/pull/62601
- https://github.com/tgstation/tgstation/pull/64982
- https://github.com/tgstation/tgstation/pull/67219

## Why It's Good For The Game

Fix conflicts with wallitems. Standardizes dirs. 

Hopefully we can standardize pixelshifting varedits in the future, but thats for another time.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

check MDB

Emissives:
 
image

</details>

## Changelog
:cl: rkz, esainane, fippefi, TastyFish
add: emissives for air alarms & request consoles
refactor: Wall mounted and directional objects have undergone major internal simplification. Please report anything unusual!
fix: You can no longer stack an indefinite amount of Intercoms on the same wall.
fix: Wall mounted items on top of the wall now consistently check against other items on top of the wall, and items coming out of the wall now consistently check against other items coming out of the wall.
fix: The various directional pixel offsets within an APC, Fire Extinguisher Cabinet, Intercom, or Newscaster have been made consistent with each other.
fix: The pixel offsets of Intercoms, Fire Alarms, Fire Extinguisher Cabinets, Flashers, and Newscasters have been made consistent between roundstart and constructed instances.
/:cl:

### CODE REVIEW

**Comment by MarkusLarsson421** [comment]
> Hopefully we can standardize pixelshifting varedits in the future, but thats for another time.

Made a PR for this, but it requires going over every single map one by one, updating the UpdatePath script, or making your own script to change all of them.

File: code/__DEFINES/power/apc_defines.dm [hunk 1]
```diff
@@ -14,9 +14,9 @@
 /// The APCs cover is missing.
 #define APC_COVER_REMOVED 2
 
-// !NOT IMPLEMENTED! APC visuals 
-/// Pixel offset of the APC from the floor turf 
-//#define APC_PIXEL_OFFSET 25
+// APC visuals
+/// Pixel offset of the APC from the floor turf
+#define APC_PIXEL_OFFSET 25
```
Comment (PestoVerde322) [inline code/__DEFINES/power/apc_defines.dm:19]: Not a power of 2, round it off to 24 possibly or 26

File: code/game/machinery/firealarm.dm [hunk 2]
```diff
@@ -11,6 +11,7 @@
 	icon = 'icons/obj/monitors.dmi'
 	icon_state = "fire_bitem"
 	result_path = /obj/machinery/firealarm
+	pixel_shift = 26
```
Comment (PestoVerde322) [inline code/game/machinery/firealarm.dm:14]: Previous pixel "shifting" was 24, why change it to 26?

File: code/game/machinery/newscaster/newscaster_machine.dm [hunk 3]
```diff
@@ -954,6 +956,6 @@
 	icon_state = "newscaster"
 	custom_materials = list(/datum/material/iron=14000, /datum/material/glass=8000)
 	result_path = /obj/machinery/newscaster
-	pixel_shift = -32
+	pixel_shift = 30
```
Comment (PestoVerde322) [inline code/game/machinery/newscaster/newscaster_machine.dm:959]: Same as before, previous 32, now 30

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 4]
```diff
@@ -1044,6 +1041,7 @@ DEFINE_BUFFER_HANDLER(/obj/machinery/turretid)
 	icon_state = "apc"
 	result_path = /obj/machinery/turretid
 	custom_materials = list(/datum/material/iron=MINERAL_MATERIAL_AMOUNT)
+	pixel_shift = 29
```
Comment (PestoVerde322) [inline code/game/machinery/porta_turret/portable_turret.dm:1044]: Also power of 2 (knock it down to 28?); however in this case is a wallmount that magically from an APC frame becomes a Turret control panel, may require a new icon for it.

**Review by PestoVerde322 — CHANGES_REQUESTED** [review]

File: code/__DEFINES/power/apc_defines.dm (hunk 1 above)
Comment (Tsar-Salat) [reply code/__DEFINES/power/apc_defines.dm:19]: Is this meant to be a suggestion comment

Comment (PowerfulBacon) [reply code/__DEFINES/power/apc_defines.dm:19]: 24 and 26 are even numbers, not powers of 2. Why would it matter anyway?

File: code/game/machinery/newscaster/newscaster_machine.dm (hunk 3 above)
Comment (Tsar-Salat) [reply code/game/machinery/newscaster/newscaster_machine.dm:959]: No. The previous is -32. We added directional support for this item, so the pixel shift is not the same. I changed it to accomodate .dmi changes and for the mapping helper macro

File: code/game/machinery/firealarm.dm (hunk 2 above)
Comment (Tsar-Salat) [reply code/game/machinery/firealarm.dm:14]: I thought it looked off.

image

I think they should be just below air alarms, but a bit higher than they were. Just a consideration for possible wallening in the future.

File: code/game/machinery/porta_turret/portable_turret.dm (hunk 4 above)
Comment (Tsar-Salat) [reply code/game/machinery/porta_turret/portable_turret.dm:1044]: > however in this case is a wallmount that magically from an APC frame becomes a Turret control panel, may require a new icon for it.

I dont see what my pr has to do with that.

File: code/__DEFINES/power/apc_defines.dm (hunk 1 above)
Comment (Tsar-Salat) [reply code/__DEFINES/power/apc_defines.dm:19]: I dont think it being even matters. If it looks fine in my opinion, I think thats enough. I did a bunch of manual edits to fix the screwups of the script, I think its in a good place.

If you see any glaring issues where wallitems dirs or pixelshifts look bad through MDB, sure I can fix it but Im not going to act on arbitrary numbers.

**Review by PowerfulBacon — APPROVED** [review]

**Review by PestoVerde322 — APPROVED** [review]
Aright, mapwise seems all good, no artifacts and whatnot, codewise seems all good to go, so approved.

Now notes for the future that would be out of scope for this PR:
- Many fixtures themselves don't use helpers yet so mappers will have to convert them from normal to directional
- While looking around in SDMM, I've noticed that many wallmonunts are actually not wallmounts, therefore they will need to be converted.
- Clock wallmounts still use the old system for directionals, that can be addressed either now or later, but i think that's ok due to the rarity of those.

That's all i can say, finally we can have a better standard for mapping stuff.

**Pull request merged**